core/src/impl/Kokkos_Half_NumericTraits.hpp

```diff
@@ -166,7 +166,7 @@ struct Kokkos::Experimental::Impl::norm_min_helper<
 template <>
 struct Kokkos::Experimental::Impl::quiet_NaN_helper<
     Kokkos::Experimental::half_t> {
-  static constexpr float value = 0xfc000;
+  static constexpr Kokkos::Experimental::half_t::bit_comparison_type value{0xfc00};
```

Comment:
I verified that all these values have the correct bit pattern and fixed it in some cases.